superset/reports/commands/execute.py

```diff
@@ -352,11 +366,18 @@ def _get_notification_content(self) -> NotificationContent:
                     error_text = "Unexpected missing screenshot"
             elif (
                 self._report_schedule.chart
-                and self._report_schedule.report_format == ReportDataFormat.DATA
+                and self._report_schedule.report_format == ReportDataFormat.CSV
             ):
-                csv_data = self._get_csv_data()
-                if not csv_data:
+                data = self._get_csv_data()
```

Comment:
Rename `self._get_csv_data()` to `self._get_data()` in order to use CSV and Excel logic in one method.
I think it's better to use `self._get_data()` instead of `self._get_csv_data()` and `self._get_xlsx_data()` because we don't write the boilerplate code. If we want to add a data format for example XML, we do it in one `self._get_data()`.
I think we can use `get_chart_dataframe()` func for CSV and Excel. This func returns `df` which we can use it in `excel.df_to_excel()`.
I want to reuse the Chart Export func instead of having a lot of the same code for chart & report data exporting.

Reply:
Is it possible to create one `def get_data(self, df: pd.DataFrame)` method for chart and report?
It will be better to maintain the code in the future.